.github/actions/backport/action.yml

```diff
@@ -0,0 +1,79 @@
+name: Backport
+
+# A composite action that allows Backporting via GitHub Actions (e.g. to be driven based on labels etc)
+
+inputs:
+  GITHUB_TOKEN:
+    required: true
+  REPO_LOCATION:
+    description: The location of the checked-out repository to work with
+    required: true
+  TARGET_BRANCH:
+    description: The branch in the repository to backport changes into
+    required: true
+  REF_TO_BACKPORT:
+    description: The reference of the commit to be backported
+    required: true
+  BACKPORT_OPTIONS:
+    description: Additional options to pass through to the tool
+    required: false
+
+env:
+  # Not possible to set this as a default
+  # https://github.com/orgs/community/discussions/46670
+  shell: bash
+
+runs:
+  using: composite
+  steps:
+    - uses: actions/checkout@v4
+      with:
+        repository: hazelcast/backport
+        # Hardcoded as composite actions don't support env vars properly
+        # https://github.com/orgs/community/discussions/51280
+        path: backport
+
+    - name: Checkout maintenance branch and cherry-pick
+      shell: ${{ env.shell }}
+      working-directory: ${{ inputs.REPO_LOCATION }}
+      run: |
+        # Git metadata is required but not available out-of-the-box, inherit from the action
+        git config user.name "${GITHUB_ACTOR}"
+        git config user.email "${GITHUB_ACTOR}@users.noreply.github.com"
+
+        # Add "upstream" remote as checkout action doesn't include by default
+        git remote add upstream "${{ github.event.repository.clone_url }}"
+        git fetch --all
+
+        backport_target_branch=upstream/"${{ inputs.TARGET_BRANCH }}"
+        echo "::debug::Running backport script to backport "${{ inputs.REF_TO_BACKPORT }}" into \"${backport_target_branch}\""
+
+        ${GITHUB_WORKSPACE}/backport/backport \
+          "${{ inputs.REF_TO_BACKPORT }}" \
+          "${backport_target_branch}" \
+          --non-interactive \
+          ${{ inputs.BACKPORT_OPTIONS }}
+      env:
+        GH_TOKEN: ${{ github.token }}
+
+    - name: Report errors
+      shell: ${{ env.shell }}
+      if: failure()
+      run: |
+        echo ":error::Error running action"
+        echo "::group::Troubleshooting Information:"
+        echo "- Repositories' GitHub action configuration - $GITHUB_SERVER_URL/$GITHUB_REPOSITORY/settings/actions"
+        echo "- Action does not run when triggered from a forked repo's PR?"
```

Comment:
I don't think so, but good to raise.

Specifically, the questionable ones only seem to apply to private repos anyway, e.g. compare:
- https://github.com/hazelcast/hz-docs/settings/actions
- https://github.com/hazelcast/hazelcast-mono/settings/actions

Reply:
Ok. I don't even see section `Fork pull request workflows` in hz-docs likely because its public
Wonder if its worth adding a warning to say 'you need to be careful when changing settings to ensure you don't expose secrets'. Not critical so will resolve but feel free to add